lib/airtable.ts

```diff
@@ -0,0 +1,145 @@
+
+import Airtable from 'airtable';
+require('dotenv').config();
+
+const base = new Airtable({ apiKey: process.env.AIRTABLE_API_KEY }).base('appUW06bs08YxzVDM')
+
+const getIdea = async ({ id }) => {
+  try {
+    // 1. Find Idea by by ID   
+    const record = await base('Ideas').find(id)
+    // 2. Fetch all Actions (with titles and type) of this idea
+    const filterStr = record.fields['Actions'].reduce((acc, recID) => {
+      return `${acc}RECORD_ID() = '${recID}', `;
+    }, 'OR(').slice(0, -2) + ')';
+
+    const actionRecords = await base('Actions').select({
+      view: 'Grid view',
+      filterByFormula: filterStr,
+      fields: ['Action Title', 'Action Type', 'Count', 'By Users'],
+    }).firstPage();
+    return {
+      record,
+      actionRecords
+    }
+  } catch (err) {
+    console.log(err);
+  }
+  return {
+    record: [] as any,
+    actionRecords: [] as any,
+  }
+}
+
+const getAllIdeas = async () => {
+  try {
+    const records = await base('Ideas').select({
+      view: 'Grid view',
+      pageSize: 5,
+      filterByFormula: 'AND({Approval} = "Approved")',
+      sort: [{ field: "Support Count", direction: "desc" }],
+    }).firstPage();
+    return {
+      records
+    }
+  } catch (err) {
+    console.log(err);
+  }
+  return {
+    records: []
+  }
+}
+
+const getUserOrRegister = async ({ user }) => {
+  const userRecs = await base('Users').select({
+    view: 'Grid view',
+    filterByFormula: `{User Id} = '${user.id}'`, //use user id here as username might change
+  }).firstPage();
+  return userRecs.length === 0 ?
+    await base('Users').create({
+      "Username": user.username,
+      "User Id": String(user.id),
+    }, { typecast: true })
+    : userRecs[0]
+}
+
+const getUserLastActionRec = async ({ userRecord, actionRecords }) => {
+  const eachAction = await actionRecords.find(async eachAction => {
+    const userRecIdArr = eachAction.fields['By Users'] || [];
+    if (userRecIdArr.includes(userRecord.id)) {
+      return eachAction
+    }
+  })
+  return eachAction
+};
+
+const addUsersOnAction = async ({ selectedActionId, userRecord, existingSupporters }) => {
+  if (existingSupporters.includes(userRecord.id)) return
```

Comment:
Sorry if it's a silly question:
so it's more preferred to have no `;` after return nowadays?